libraries/chain/include/eosio/chain/webassembly/eos-vm-oc/eos-vm-oc.hpp

```diff
@@ -31,6 +31,7 @@ struct code_descriptor {
    size_t code_begin;
    eosvmoc_optional_offset_or_import_t start;
    unsigned apply_offset;
+   std::optional<unsigned> call_offset;
```

Comment:
I think this might cause a problem for existing code caches that may or may not be properly handled on start up. Changing the `code_descriptor` layout probably requires indicating something higher up (like at the header) that the format has been changed and the cache must be destroyed and recreated at start up.

Reply:
No I don't believe that will help. The serialized format of `code_descriptor` has changed, and it isn't versioned. I think you need to change
https://github.com/AntelopeIO/spring/blob/8496995f58641fb19cff38966b92d0df10314813/libraries/chain/webassembly/runtimes/eos-vm-oc/code_cache.cpp#L21
So that the cache file is completely removed and recreated

Reply:
Thanks! I missed that.